README.md

```diff
@@ -92,40 +92,73 @@ You'll also need to create a private key for the service account (it should down
     ```
 
-    - Make sure to use https:// URLs instead of http:// ones
-    - Some of the settings will need to be verified before they get save
-      - This means that you will need to update your `.env` file with the settings from your Slack app
-      - Reload your server for the new env vars to apply and resend the verification payloads
-      - You will have to do this with multiple settings, thus, you will have to repeat reloading your server as you add new variables
-    - On your new Slack workspace begin a conversation with the bot
-    - You should see your localhost app respond with 200 status code
-    - Congratulations!
+    - Some of the settings will need to be verified before they get saved
+    - This means that you will need to update your `.env` file with the settings from your Slack app
+    - Reload your server for the new env vars to apply and resend the verification payloads
+    - You will have to do this with multiple settings, thus, you will have to repeat reloading your server as you add new variables
+
+1.  Create a new GitHub repository at `<YOUR_GITHUB_USERNAME>/eng-pipes-dev`.
+1.  Create a [new GitHub App](https://github.com/settings/apps/new)
+    - Set the webhook to your ngrok tunnel with the GH route (e.g. `<NGROK_INSTANCE>/webhooks/github`)
+    - Place the secrets in your `.env` (see [Setup Secrets](#setup-secrets) below)
+    - Go to the `Permissions & events` sidebar menu entry of the GitHub app configuration, and grant maximum non-`Admin` access (`Read and write` where possible, `Read only` everywhere else) for every line in `Repository permissions`
+    - For `Organization permissions`, grant `Read and write` for `Members` and `Projects`
+    - In the `Subscribe to events` section, check every possible box
+1.  Go to `Install App` in the sidebar menu of the GitHub app configuration, and install the app for your GitHub account.
+    - When prompted, choose `Only select repositories`, and pick only the `eng-pipes-dev` repository you created in step 4
+1.  Follow the steps of the "Development & tests" section below to get the server running.
+    - It will fail if you don't have all the correct env variables defined
+1.  Verify that the Slack -> eng-pipes server pipeline works
+
+    - On your new Slack workspace, send a message to the bot
+    - You should see your localhost app respond with a 200 status code:
+
+    ![success!](/docs/successful_slack_message.png 'Successful Slack message forwarding')
 
-NOTE: ngrok gives you a [localhost interface](http://127.0.0.1:4040/inspect/http) to see events coming and to replay them.
+1.  Verify that the GitHub -> eng-pipes server pipeline works
+
+    - In your `eng-pipes-dev` GitHub repository, create a new issue
+    - You should see your localhost app response with a 200 status code:
+
+    ![success!](/docs/successful_github_event.png 'Successful GitHub webhook reception')
+
+Congratulations, you're all setup!
+
+### Quality-of-life tips
```

Comment:
should we add something in here about the `DRY_RUN` variable? Since it gets rid of the bad auth requests to BigQuery

Reply:
YEASSSSS

Reply:
Done.